client/my-sites/plans-grid/components/actions.tsx

```diff
@@ -312,10 +302,10 @@ const LoggedInPlansFeatureActionButton = ( {
 		return (
 			<Button
 				className={ classes }
-				href={ currentPlanManageHref }
-				disabled={ ! currentPlanManageHref }
+				disabled={ ! planActionOverrides?.currentPlan?.callback }
+				onClick={ planActionOverrides?.currentPlan?.callback }
 			>
-				{ canUserManageCurrentPlan ? translate( 'Manage plan' ) : translate( 'View plan' ) }
+				{ planActionOverrides?.currentPlan?.text }
```

Comment:
Thanks for working out these changes! 

My only concern here is `planActionOverrides?.currentPlan` feels like something of a "required" property, otherwise we'd still be rendering a button but with no text. That will probably never be the case technically, but semantically/conceptually that is the case. 

I think the above is fine, especially if we want to keep the two texts living close to each other. but let's maybe make it more explicit that this is what we are doing. I would maybe create a separate "default" case (the button with no functionality) and conditionally render it if no `planActionOverrides?.currentPlan` exists. That way it's a little more clear that we are "replacing" something (instead of "injecting" things into it).

I will explore a little later and send a commit if I come with something :-)

Reply:
@chriskmnds I have added a `default` case! Can you have another look?

https://github.com/Automattic/wp-calypso/pull/83976/commits/91f88ec491418b180b59bf14488d263623bf74d8